### PR TITLE
Update for SparseWeights mask change

### PR DESCRIPTION
**Pull request by mrcslws**
Makes ImagenetExperiment auto-update old checkpoints

Depends on https://github.com/numenta/nupic.torch/pull/70

### CODE REVIEW

File: tests/unit/frameworks/pytorch/restore_utils_test.py [hunk 1]
```diff
@@ -139,20 +139,22 @@ def test_load_full(self):
...
-        # Check output through the lower network.
-        out = lower_forward(model, self.in_1)
-        num_matches = out.isclose(self.out_lower, atol=1e-2).sum().item()
-        self.assertEqual(num_matches, 1337)  # some correct
-
-        # Check output through the lower network.
-        out = upper_forward(model, self.in_2)
-        num_matches = out.isclose(self.out_upper, atol=1e-2).sum().item()
-        self.assertEqual(num_matches, 1)  # some correct
+        # These exact numbers are random.
```
Comment (lscheinkman) [inline tests/unit/frameworks/pytorch/restore_utils_test.py:142]: Instead of commenting out the test could you add a "skip" comment just like you added to `test_checkpoint_backward_compatibility`?

File: tests/unit/frameworks/pytorch/restore_utils_test.py [hunk 2]
```diff
@@ -193,15 +199,17 @@ def test_load_nonlinear(self):
...
```
Comment (lscheinkman) [inline tests/unit/frameworks/pytorch/restore_utils_test.py:202]: Instead of commenting out the test could you add a "skip" comment just like you added to `test_checkpoint_backward_compatibility`?

File: tests/unit/frameworks/pytorch/restore_utils_test.py [hunk 3]
```diff
@@ -212,10 +220,13 @@ def test_load_nonlinear(self):
         num_matches = out.isclose(self.out_lower, atol=1e-2).sum().item()
         self.assertEqual(num_matches, 2048)  # all correct
 
-        # Check output through the lower network.
-        out = upper_forward(model, self.in_2)
-        num_matches = out.isclose(self.out_upper, atol=1e-2).sum().item()
-        self.assertEqual(num_matches, 1)
+        # These exact numbers are random, the test should test for something
```
Comment (lscheinkman) [inline tests/unit/frameworks/pytorch/restore_utils_test.py:223]: Instead of commenting out the test could you add a "skip" comment just like you added to `test_checkpoint_backward_compatibility`?

File: tests/unit/frameworks/pytorch/restore_utils_test.py [hunk 4]
```diff
@@ -229,24 +240,30 @@ def test_load_linear(self):
...
-        # Check output through the lower network.
-        out = lower_forward(model, self.in_1)
-        num_matches = out.isclose(self.out_lower, atol=1e-2).sum().item()
-        self.assertEqual(num_matches, 1337)  # some correct
-
-        # Check output through the lower network.
-        out = upper_forward(model, self.in_2)
-        num_matches = out.isclose(self.out_upper, atol=1e-2).sum().item()
-        self.assertEqual(num_matches, 1)  # some correct
+        # These exact numbers are random, the test should test for something
```
Comment (lscheinkman) [inline tests/unit/frameworks/pytorch/restore_utils_test.py:243]: Instead of commenting out the test could you add a "skip" comment just like you added to `test_checkpoint_backward_compatibility`?

File: tests/unit/frameworks/pytorch/restore_utils_test.py [hunk 5]
```diff
@@ -229,24 +240,30 @@ def test_load_linear(self):
...
 
         # Restore full model.
         model = load_multi_state(model, restore_linear=self.checkpoint_path)
         model.eval()
 
-        # Check output through the lower network.
-        out = lower_forward(model, self.in_1)
-        num_matches = out.isclose(self.out_lower, atol=1e-2).sum().item()
-        self.assertEqual(num_matches, 1337)  # some correct
+        # These exact numbers are random, the test should test for something
```
Comment (lscheinkman) [inline tests/unit/frameworks/pytorch/restore_utils_test.py:260]: Instead of commenting out the test could you add a "skip" comment just like you added to `test_checkpoint_backward_compatibility`?

File: tests/unit/frameworks/pytorch/restore_utils_test.py (hunk 1 above)
Comment (mrcslws) [reply tests/unit/frameworks/pytorch/restore_utils_test.py:142]: But some parts of these tests are still valid.

Comment (lscheinkman) [reply tests/unit/frameworks/pytorch/restore_utils_test.py:142]: In that case we can either break these testes into 2 different parts or remove the commented out code if is no longer relevant. 
It is just hard to maintain commented out code.
